src/main/java/frc/robot/commands/drive/Drive.java

```diff
@@ -15,24 +20,60 @@
      */
     private boolean sideFlipped = false;
 
+    private double pitchDegrees;
+
     public Drive() {
         requires(Drivetrain.getInstance());
     }
 
+    private void correctForPitch(double[] stickSpeeds) {//x-tip
+        //if its over 45 there's no point in correcting it
+        if (Math.abs(pitchDegrees) < 45) {
+            //correctionFactor keeps the tilt correction within a certain threshold so it doesn't correct too much
+            double correctionFactor = (RobotMap.DRIVE.MAX_CORRECTION_FACTOR - RobotMap.DRIVE.MIN_CORRECTION_FACTOR) / (45 - RobotMap.DRIVE.PITCH_THRESHOLD_DEGREES);
+
+            double correctionOffset = correctionFactor * (pitchDegrees - RobotMap.DRIVE.PITCH_THRESHOLD_DEGREES);
+            stickSpeeds[0] += correctionOffset;
+            stickSpeeds[1] += correctionOffset;
+            RobotMath.normalize(stickSpeeds);
+        }
+    }
+
+    private void updatePitchStatus() {
+        this.pitchDegrees = NavX.getInstance().getPitchDegrees();
+    }
+
+    private void displayTippiness(){
+        boolean forwardProblem = NavX.getInstance().getPitchDegrees() > RobotMap.DRIVE.PITCH_THRESHOLD_DEGREES;
+        boolean backwardProblem = NavX.getInstance().getPitchDegrees() < -RobotMap.DRIVE.PITCH_THRESHOLD_DEGREES;
+
+        SmartDashboard.putBoolean("Tipping Forward", forwardProblem);
+        SmartDashboard.putBoolean("Tipping Backward", backwardProblem);
+    }
+
     @Override
     protected void initialize() {
         setCamera("RPI");
     }
 
     @Override
     protected void execute() {
-        double leftStick = powerCurve(OI.getInstance().getLeftY());
-        double rightStick = powerCurve(OI.getInstance().getRightY());
+        double stickSpeeds[] = {RobotMath.powerCurve(OI.getInstance().getLeftY()), RobotMath.powerCurve(OI.getInstance().getRightY())};
+        //double leftStick = RobotMath.powerCurve(OI.getInstance().getLeftY());
+        //double rightStick = RobotMath.powerCurve(OI.getInstance().getRightY());
+
+        updatePitchStatus();
+        if(!Drivetrain.getInstance().getIsAntiTipDisabled()) {
+            if (NavX.getInstance().getPitchDegrees() > RobotMap.DRIVE.PITCH_THRESHOLD_DEGREES) {
+                this.correctForPitch(stickSpeeds);
+            }
+        }
+        this.displayTippiness();
```

Comment:
Repeated calls to `SmartDashboard.putThing` have been known to cause problems this year. Possibly due to limited bandwidth, but at any rate, be cautious of these. (The driver can see the robot tip most of the time... )